packages/framework/fluid-framework/src/index.ts

```diff
@@ -54,13 +54,21 @@ export type {
 	ReplaceIEventThisPlaceHolder,
 	FluidObject, // Linked in doc comment
 	FluidObjectProviderKeys, // Used by FluidObject
+	// eslint-disable-next-line import/export
+	Listeners,
+	// eslint-disable-next-line import/export
+	IsListener,
+	// eslint-disable-next-line import/export
+	Listenable,
+	// eslint-disable-next-line import/export
+	Off,
```

Comment:
@microsoft/fluid-cr-api (API council), should these long-term be exported from fluid-framework?
I would say they should not be as they aren't something that we expect customers to reference directly. So, we keep them out and keep this surface smaller. Sound, right? They can always get them directly from core-interfaces if they need them.

Reply:
I could see a customer using `Off` (e.g. they want to store a deregistration function, so they type it as `Off` rather than `() => void`). I could also see a customer using `Listenable` - for example, when writing a class/function which accepts a `Listenable` as an argument, and then subscribes to some events.

I wouldn't expect a user to use `Listeners` or `IsListener` directly under normal circumstances. Maybe it would be even be appropriate to mark them as @system, if it's not too late to do that.

Reply:
Our approach so far has been to export all visible types, otherwise I think customers probably struggle to type certain usage.  I don't think it'd be unreasonable for customers to reference these directly.  Especially things like `Off`, where customers are likely to maintain it as a private member for later listener removal.

I mention in my other comment that we should drop the export from the tree package though, as I don't know what the duplicated export actually does.

Reply:
> I wouldn't expect a user to use Listeners or IsListener directly under normal circumstances. Maybe it would be even be appropriate to mark them as `@system`, if it's not too late to do that.

I believe these have been tagged `@public` and published in `fluid-framework` since 2.0.0, right?  So restricting them now would be a breaking change.

Reply:
We _could_ mark the core-interfaces ones `@system` and deprecate the tree and fluid-framework exposure. But maybe just make a remark or private remark for now.
(I wonder how many folks with <common tag> usernames regret they choices per all our `@` mentions. Usually I only notice after.)
I think that fluid-framework is missing a lot of types - that our settings permit missing things that are from other packages. Happy to be wrong.
`Off` seemed most likely to me, but didn't see that useful. I think I see the `Listenable` use case - would be good to have an example.

Reply:
I would still err on the side of exposing just what we think is useful in the future. Since we have exposure now and we think some might be useful, then we can keep it.